src/network.rs

```diff
@@ -240,21 +269,44 @@ impl NetworkBehaviourEventProcess<orderbook::BehaviourOutEvent> for Nectar {
                 response_channel,
                 order_id,
             } => {
+                let taker = Taker::new(taker_peer_id);
+                let ongoing_trade_with_taker_exists = match self.active_takers.contains(&taker) {
+                    Ok(res) => res,
+                    Err(e) => {
+                        tracing::warn!(
```

Comment:
This is more appropriate as an error because you cannot process more orders due to a DB issue.

Reply:
Good point, thanks for reviewing the details :relaxed: